cypress/utils/utils.ts

```diff
@@ -1212,15 +1214,15 @@ export function selectUserPerspective(userType: string): void {
 }
 
 export function selectWithinModal(selector: string): void {
-    cy.get(modal).within(() => {
-        click(selector);
-    });
+    clickWithin(modal, selector);
 }
 
 export function clickWithin(parent, selector: string): void {
-    cy.get(parent, { timeout: 30 * SEC }).within(() => {
-        click(selector);
-    });
+    cy.get(parent, { timeout: 30 * SEC })
+        .eq(0)
+        .within(() => {
+            click(selector);
+        });
```

Comment:
a little suggestion, the function can be generic with a default value, we can pass a parameter 
`export function clickWithin(parent, selector: string, nthPosition = 0): void {`
and it's use can be as follows:
`cy.get(parent, { timeout: 30 * SEC })
        .eq(nthPosition)
        .within(() => {
            click(selector);
        });`

Reply:
Good suggestion @midays . We can do that as an improvement once we have all tests stabilized.